packages/form-core/src/FieldApi.ts

```diff
@@ -696,3 +705,46 @@ function getErrorMapKey(cause: ValidationCause) {
       return 'onChange'
   }
 }
+
+export function deepEqual<T>(objA: T, objB: T): boolean {
```

Comment:
This function is a modified version of `shallow` from `tanstack/store`: https://github.com/TanStack/store/blob/be32993f5224c8e60ce6c3fe6674ec5907713f80/packages/react-store/src/index.ts#L27-L55